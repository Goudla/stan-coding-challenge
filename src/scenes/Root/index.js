import React, { Component } from 'react';
import { Switch, Route, Redirect } from 'react-router';
import { FaFacebookF, FaTwitter, FaInstagram } from 'react-icons/fa';

// Components
import Button from '../../components/Button';
import ButtonLink from '../../components/ButtonLink';
import Container from '../../components/Container';
import FlexGrid from '../../components/FlexGrid';
import Footer from '../../components/Footer';
import FooterCopyright from '../../components/FooterCopyright';
import FooterExternalLinks from '../../components/FooterExternalLinks';
import FooterLink from '../../components/FooterLink';
import FooterLinks from '../../components/FooterLinks';
import FooterSocialLink from '../../components/FooterSocialLink';
import FooterSocialLinks from '../../components/FooterSocialLinks';
import FooterStoreLink from '../../components/FooterStoreLink';
import FooterStoreLinks from '../../components/FooterStoreLinks';
import IconAppStore from '../../components/IconAppStore';
import IconPlayStore from '../../components/IconPlayStore';
import IconWindowsStore from '../../components/IconWindowsStore';
import Navbar from '../../components/Navbar';
import NavbarColLeft from '../../components/NavbarColLeft';
import NavbarColRight from '../../components/NavbarColRight';
import NavbarHeading from '../../components/NavbarHeading';
import PageContent from './components/PageContent';
import PageFooter from './components/PageFooter';
import PageHeader from './components/PageHeader';
import PageWrapper from './components/PageWrapper';

// Scenes
import Entries from '../Entries';
import Home from '../Home';

/**
 * The root of the application
 * @extends Component
 */
class Root extends Component<*> {
  render() {
    return (
      <PageWrapper>
        <PageHeader>
          <Navbar>
            <Container>
              <FlexGrid>
                <NavbarColLeft>
                  <NavbarHeading>DEMO Streaming</NavbarHeading>
                </NavbarColLeft>
                <NavbarColRight>
                  <ButtonLink>Log in</ButtonLink>
                  <Button>Start your free trial</Button>
                </NavbarColRight>
              </FlexGrid>
            </Container>
          </Navbar>
        </PageHeader>
        <PageContent>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route
              exact
              path="/series"
              render={() => <Entries title="Series" programType="series" />} />
            <Route
              exact
              path="/movies"
              render={() => <Entries title="Movies" programType="movie" />} />
            <Redirect from='/*' to="/" />
          </Switch>
        </PageContent>
        <PageFooter>
          <Footer>
            <Container>
              <FooterLinks>
                <FooterLink>Home</FooterLink>
                <FooterLink>Terms and Conditions</FooterLink>
                <FooterLink>Privacy Policy</FooterLink>
                <FooterLink>Collection Statement</FooterLink>
                <FooterLink>Help</FooterLink>
                <FooterLink>Manage Account</FooterLink>
              </FooterLinks>
              <FooterCopyright>Copyright Â© 2016 DEMO Streaming. All Rights Reserved.</FooterCopyright>
              <FooterExternalLinks>
                <FooterSocialLinks>
                  <FooterSocialLink>
                    <FaFacebookF size="30" />
                  </FooterSocialLink>
                  <FooterSocialLink>
                    <FaTwitter size="30" />
                  </FooterSocialLink>
                  <FooterSocialLink>
                    <FaInstagram size="30" />
                  </FooterSocialLink>
                </FooterSocialLinks>
                <FooterStoreLinks>
                  <FooterStoreLink>
                    <IconAppStore />
                  </FooterStoreLink>
                  <FooterStoreLink>
                    <IconPlayStore />
                  </FooterStoreLink>
                  <FooterStoreLink>
                    <IconWindowsStore />
                  </FooterStoreLink>
                </FooterStoreLinks>
              </FooterExternalLinks>
            </Container>
          </Footer>
        </PageFooter>
      </PageWrapper>
    );
  }
}

export default Root;
