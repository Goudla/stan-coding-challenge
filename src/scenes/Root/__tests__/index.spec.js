import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Scenes
import Root from '../';

describe('<Root />', () => {
  it('Should render', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Root />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
