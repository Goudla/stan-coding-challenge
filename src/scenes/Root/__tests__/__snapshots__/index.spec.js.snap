// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> Should render 1`] = `
<PageWrapper>
  <PageHeader>
    <Navbar>
      <Container>
        <FlexGrid>
          <NavbarColLeft>
            <NavbarHeading>
              DEMO Streaming
            </NavbarHeading>
          </NavbarColLeft>
          <NavbarColRight>
            <ButtonLink>
              Log in
            </ButtonLink>
            <Button>
              Start your free trial
            </Button>
          </NavbarColRight>
        </FlexGrid>
      </Container>
    </Navbar>
  </PageHeader>
  <PageContent>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        exact={true}
        path="/series"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/movies"
        render={[Function]}
      />
      <Redirect
        from="/*"
        to="/"
      />
    </Switch>
  </PageContent>
  <PageFooter>
    <Footer>
      <Container>
        <FooterLinks>
          <FooterLink>
            Home
          </FooterLink>
          <FooterLink>
            Terms and Conditions
          </FooterLink>
          <FooterLink>
            Privacy Policy
          </FooterLink>
          <FooterLink>
            Collection Statement
          </FooterLink>
          <FooterLink>
            Help
          </FooterLink>
          <FooterLink>
            Manage Account
          </FooterLink>
        </FooterLinks>
        <FooterCopyright>
          Copyright Â© 2016 DEMO Streaming. All Rights Reserved.
        </FooterCopyright>
        <FooterExternalLinks>
          <FooterSocialLinks>
            <FooterSocialLink>
              <FaFacebookF
                size="30"
              />
            </FooterSocialLink>
            <FooterSocialLink>
              <FaTwitter
                size="30"
              />
            </FooterSocialLink>
            <FooterSocialLink>
              <FaInstagram
                size="30"
              />
            </FooterSocialLink>
          </FooterSocialLinks>
          <FooterStoreLinks>
            <FooterStoreLink>
              <IconAppStore />
            </FooterStoreLink>
            <FooterStoreLink>
              <IconPlayStore />
            </FooterStoreLink>
            <FooterStoreLink>
              <IconWindowsStore />
            </FooterStoreLink>
          </FooterStoreLinks>
        </FooterExternalLinks>
      </Container>
    </Footer>
  </PageFooter>
</PageWrapper>
`;
